docker/Dockerfile

```diff
@@ -51,18 +51,33 @@ RUN apt-get update &&apt-get install -y \
     libsgx-ra-network \
     libsgx-ra-uefi
 
-# The prebuild app
-WORKDIR /home/ehsm/ehsm-core
-COPY bin/ehsm-core/ehsm-core /home/ehsm/ehsm-core/ehsm-core
-COPY bin/ehsm-core/libenclave-ehsm-core.signed.so /home/ehsm/ehsm-core/libenclave-ehsm-core.signed.so
 
 # Build App from source
-#RUN git clone https://github.com/intel/ehsm.git \
-#    && cd ehsm \
-#    && make
+RUN git clone https://github.com/intel/ehsm.git \
+   && cd ehsm \
+   && make
+
+# Install node & npm packages
+RUN wget https://nodejs.org/dist/v16.13.0/node-v16.13.0-linux-x64.tar.xz \
+    && tar xf  node-v16.13.0-linux-x64.tar.xz \
+    && mv node-v16.13.0-linux-x64/ /usr/local/nodejs \
+    && ln -s /usr/local/nodejs/bin/node /usr/local/bin \
+    && ln -s /usr/local/nodejs/bin/npm /usr/local/bin
+
+
+# Install ehsm-service dependence packages
+WORKDIR /opt/intel/ehsm/ehsm_kms_service
+RUN cp /opt/intel/ehsm/bin/ehsm-core/*so .
+RUN npm install
```

Comment:
i'm not sure what pkgs that cmd "npm install" will try to install?

can we only install the necessary files?

Reply:
npm install downloads the required dependencies according to package.json